ground/src/main/java/com/google/android/ground/persistence/sync/SurveySyncWorker.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.ground.persistence.sync
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.Data
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import com.google.android.ground.repository.SurveyRepository
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import timber.log.Timber
+
+/** Worker responsible for syncing latest surveys and LOIs from remote server to local db. */
+@HiltWorker
+class SurveySyncWorker
+@AssistedInject
+constructor(
+  @Assisted context: Context,
+  @Assisted params: WorkerParameters,
+  private val surveyRepository: SurveyRepository
+) : Worker(context, params) {
+  private val surveyId: String? = params.inputData.getString(SURVEY_ID_PARAM_KEY)!!
```

Comment:
Why is this variable marked as both nullable and the value asserted to be non-null(!!) ?

Reply:
Good catch. This was probably due to copy/paste abuse, fixed.